ush/ICpert_scripts/DTC_ensemble_setup.sh

```diff
@@ -0,0 +1,212 @@
+#
+# Make sure that required environment variables are defind and, if 
+# applicable, have valid values.
+#
+#-----------------------------------------------------------------------
+#
+if [ -z ${do_test_run+x} ] || \
+   [ -z ${RRFS_do_stoch+x} ] || \
+   [ -z ${RRFS_do_get_obs+x} ]; then
+  print_err_msg_exit "
+One or more of the following required environment variables are undefined:
+  do_test_run = ${do_test_run:-"<undefined>"}
+  RRFS_do_stoch = ${RRFS_do_stoch:-"<undefined>"}
+  RRFS_do_get_obs = ${RRFS_do_get_obs:-"<undefined>"}
+"
+fi
+
+check_var_valid_value "do_test_run" "valid_vals_BOOLEAN"
+check_var_valid_value "RRFS_do_stoch" "valid_vals_BOOLEAN"
+check_var_valid_value "RRFS_do_get_obs" "valid_vals_BOOLEAN"
+#
+#-----------------------------------------------------------------------
+#
+# Set secondary variables based on above settings.
+#
+#-----------------------------------------------------------------------
+#
+# Variable for naming experiments depending on whether a test run is
+# being performed.
+#
+uscore_test_or_null=""
+if [ "${do_test_run}" = "TRUE" ]; then
+  uscore_test_or_null="_test"
+fi
+#
+# Predefined grid, range of cycles, forecast length, and number of (RRFS)
+# ensemble members.
+#
+date_first_cycl="20220430"
+if [ "${do_test_run}" = "TRUE" ]; then
+  predef_grid_name=${predef_grid_name:-"RRFS_CONUScompact_25km"}
+  date_last_cycl=${date_last_cycl:-"20220430"}
+  fcst_len_hrs=${fcst_len_hrs:-"6"}
+  num_RRFS_ens_members=${num_RRFS_ens_members:-"03"}
+else
+  predef_grid_name=${predef_grid_name:-"RRFS_CONUScompact_3km"}
+  date_last_cycl=${date_last_cycl:-"20220512"}
+  fcst_len_hrs=${fcst_len_hrs:-"36"}
+  num_RRFS_ens_members=${num_RRFS_ens_members:-"10"}
+fi
+#
+# Array containing all the cycles for which the RRFS analog experiment
+# will be run.
+#
+all_cycles=( $(eval echo ${date_first_cycl}00 ${date_last_cycl:0:6}{01..12}00) )
+if [ "${do_test_run}" = "TRUE" ]; then
+  all_cycles=( $(eval echo ${date_first_cycl}00 ) )
+fi
+
+echo
+echo "uscore_test_or_null = \"${uscore_test_or_null}\""
+echo "predef_grid_name = \"${predef_grid_name}\""
+echo "date_first_cycl = \"${date_first_cycl}\""
+echo "date_last_cycl = \"${date_last_cycl}\""
+echo "fcst_len_hrs = \"${fcst_len_hrs}\""
+echo "num_RRFS_ens_members = \"${num_RRFS_ens_members}\""
+echo "all_cycles = ( ${all_cycles[@]} )"
+#
+#-----------------------------------------------------------------------
+#
+# Variables needed by RRFS analog experiment.
+#
+#-----------------------------------------------------------------------
+#
+if [ "${RRFS_do_stoch}" = "TRUE" ]; then
+  RRFS_no_or_null=""
+else
+  RRFS_no_or_null="no"
+fi
+#
+# Absolute paths to the RRFS analog (either without or with stochastic
+# IC perturbations) workflow and to the GEFS individual member workflows.
+#
+RRFS_analog_expt_subdir="RRFS_ICperts_${RRFS_no_or_null}stoch${uscore_test_or_null}"
+RRFS_analog_exptdir="${expt_basedir}/${RRFS_analog_expt_subdir}"
+
```

Comment:
What is meant by "analog" here and elsewhere?

Reply:
Our RRFS runs are analogs of the real RRFS runs that GSL does.  I saw @willmayfield use this terminology, so I used it too.

Reply:
I think I picked up that usage of analog from Jeff.